src/uct/tcp/tcp_iface.c

```diff
@@ -231,20 +235,28 @@ static void uct_tcp_iface_connect_handler(int listen_fd, void *arg)
 
 ucs_status_t uct_tcp_iface_set_sockopt(uct_tcp_iface_t *iface, int fd)
 {
-    int ret;
+    int sock_buf_def = (int)UCS_CONFIG_MEMUNITS_AUTO;
+    ucs_status_t status;
 
-    ret = setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, (void*)&iface->sockopt.nodelay,
-                     sizeof(int));
-    if (ret < 0) {
-        ucs_error("Failed to set TCP_NODELAY on fd %d: %m", fd);
-        return UCS_ERR_IO_ERROR;
+    status = ucs_socket_setopt(fd, IPPROTO_TCP, TCP_NODELAY,
```

Comment:
looking at this, i don't think it makes sense to pass def_optval to ucs_socket_setopt, because the value UCS_CONFIG_MEMUNITS_AUTO does not have any meaning for socket layer. would be better to check that iface->sockopt.sndbuf != MEMUNITS_AUTO in TCP code, and not sock.c

Reply:
i think TCP layer is more suitable

Reply:
agree - done